service/src/main/kotlin/fi/espoo/evaka/reports/PlacementCountReportController.kt

```diff
@@ -0,0 +1,210 @@
+    ): PlacementCountReportResult {
+        return db.connect { dbc ->
+                dbc.read {
+                    accessControl.requirePermissionFor(
+                        it,
+                        user,
+                        clock,
+                        Action.Global.READ_PLACEMENT_COUNT_REPORT
+                    )
+                    it.setStatementTimeout(REPORT_STATEMENT_TIMEOUT)
+                    it.getPlacementCountReportRows(
+                        examinationDate,
+                        requestedProviderTypes.ifEmpty { ProviderType.values().toList() },
+                        requestedCareTypes.ifEmpty { CareType.values().toList() }
+                    )
+                }
+            }
+            .also {
+                Audit.PlacementCountReportRead.log(
+                    meta =
+                        mapOf(
+                            "examinationDate" to examinationDate,
+                            "providerTypes" to requestedProviderTypes,
+                            "careTypes" to requestedCareTypes
+                        )
+                )
+            }
+    }
+
+    private fun Database.Read.getPlacementCountReportRows(
+        examinationDate: LocalDate,
+        providerTypes: List<ProviderType>,
+        careTypes: List<CareType>
+    ): PlacementCountReportResult {
+        // language=sql
+        val sql =
+            """
+WITH child_with_age AS
+         (SELECT p.id,
+                 date_part('year', age(:examinationDate, p.date_of_birth)) AS age_in_years
+          FROM person p
+          WHERE exists(SELECT FROM placement pl WHERE pl.child_id = p.id))
+SELECT ca.id                                                AS area_id,
+       ca.name                                              AS area_name,
+       d.id                                                 AS daycare_id,
+       d.name                                               AS daycare_name,
+       count(pl.id)                                         AS placement_count,
+       count(pl.id)
+       FILTER (WHERE cwa.age_in_years > 2)                  AS placement_count_3v_and_over,
+       count(pl.id)
+       FILTER (WHERE cwa.age_in_years < 3)                  AS placement_count_under_3v,
+       COALESCE(sum(CASE
+               WHEN cwa.age_in_years > 2
+                   THEN COALESCE(sno.occupancy_coefficient, 0) * COALESCE(an.capacity_factor, 1.00)
+               ELSE COALESCE(sno.occupancy_coefficient_under_3y, 0) *
+                    COALESCE(an.capacity_factor, 1.00) END), 0) AS calculated_placements
+FROM daycare d
+         JOIN care_area ca
+              ON d.care_area_id = ca.id
+         JOIN placement pl
+                   ON pl.unit_id = d.id AND daterange(pl.start_date, pl.end_date) @> :examinationDate::date
+         JOIN child_with_age cwa
+                   ON pl.child_id = cwa.id
+         JOIN service_need sn
```

Comment:
Yes. Verified this from the requirement holder and this is the preferred filtering for placed children without a service need. The lack of a service need should be temporary and would prevent accurate scaling of the placement cost.

Reply:
There are certain types of placements that never get an explicit service need, ie. they use the default. At least in Espoo, preschool, temporary daycare and club just use defaults. This would mean that the report would never work accurately in units with these kinds of placements. Usually the default service need has the same occupancy coefficients as all the other service needs that are applicable to the placement type.

Reply:
Discussed this further with Tampere and using a left join with default sno replacement would indeed be correct here.

Reply:
Updated query added